src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,308 @@
+                "softmaxcrossentropyloss: Score and Labels must identical batch size inputs");
+        }
+
+        if((scores_shape.ndim() - 1) != label_shape.ndim())
+        {
+            MIGRAPHX_THROW(
+                "softmaxcrossentropyloss: Score and Labels must contain identical K-Dimensions");
+        }
+
+        bool is_k_dim  = (scores_shape.ndim() > 3);
+        size_t batch_size = scores_shape.lens().at(0);
+        size_t class_size = scores_shape.lens().at(1);
+
+        auto loss_tensor = info.add_instruction(
+            migraphx::make_op("convert", {{"target_type", scores_shape.type()}}), labels);
+
+        // Default weights will always be 1
+        auto weights = info.add_literal(
+            migraphx::literal(migraphx::shape(scores_shape.type(), {1}, {0}), {1}));
+        weights = info.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", {class_size}}}), weights);
+
+        // Get optional input weights (Used for mean reduction)
+        bool has_weights = (args.size() > 2);
+        if(has_weights)
+        {
+            weights = args.at(2);
+            auto weights_shape = weights->get_shape();
+
+            if(weights_shape.lens()[0] != scores_shape.lens()[1])
+            {
+                MIGRAPHX_THROW("softmaxcrossentropyloss: Invalid weight vector shape. Weight must "
+                               "contain weight for each class");
+            }
+
+            if(not contains(supported_score_weight_types, weights_shape.type()))
+            {
+                MIGRAPHX_THROW(
+                    "softmaxcrossentropyloss: weight must be either half, float, or double type");
+            }
+
+            if(weights_shape.type() != scores_shape.type())
+            {
+                MIGRAPHX_THROW(
+                    "softmaxcrossentropyloss: Weight and Scores inputs must be the same type");
+            }
+        }
+
+        // adjust weights based on ignore index is that's set
+        if(has_ignore_index)
+        {
+            auto weights_shape = weights->get_shape();
+            std::vector<float> zero_val_vect(weights_shape.elements(), 0);
+            auto zero_val = info.add_literal(migraphx::literal(weights_shape, zero_val_vect));
+            weights       = info.add_instruction(
+                migraphx::make_op("scatter_none", {{"axis", 0}}), weights, ignore_index, zero_val);
+        }
+
+        if(is_k_dim)
+        {
+            auto lens = label_shape.lens();
+            size_t d =
+                std::accumulate(lens.rbegin(), lens.rend() - 1, 1, std::multiplies<size_t>());
+            scores = info.add_instruction(
+                migraphx::make_op("reshape", {{"dims", {batch_size, class_size, d}}}), scores);
+        }
+
+        // Offload calculation of log(Softmax(scores)) for the input before we perform cross entropy
+        // loss calculation
+        auto softmax_scores = info.add_instruction(migraphx::make_op("softmax"), scores);
```

Comment:
You need to do softmax on axis 1.

Reply:
Ah Agreed. Looks like default is -1 thus for the 2d case this would always work

Reply:
Fixed